### DISCUSSION TITLE
Expose common methods in trait

### DISCUSSION THREAD

**Discussion by Roms1383** [tesselode/kira#78] [Ideas]
Hello!

I use `kira` to develop [audioware](https://github.com/cyb3rpsych0s1s/audioware) (a game native plugin to extend Cyberpunk 2077 audio capabilities, since its current audio engine methods are not exposed for the most part), and I'm wondering for example why common methods like e.g. [StaticSoundData::duration](https://docs.rs/kira/latest/kira/sound/static_sound/struct.StaticSoundData.html#method.duration) and [StreamingSoundData::duration](https://docs.rs/kira/latest/kira/sound/streaming/struct.StreamingSoundData.html#method.duration) are not exposed into their common trait [SoundData](https://docs.rs/kira/latest/kira/sound/trait.SoundData.html) (or even another one).

I'd like to know if the decision is intentional, before opening a PR to do so.

Thanks :)

**Comment by tesselode** [comment]
Despite the name, `SoundData` does not necessarily represent a finite sound with a known duration. It could also be an infinite sound. So not every valid `SoundData` would be able to implement a `duration` method.

**Comment by Roms1383** [comment]
well it's just a suggestion around ergonomics : `duration` is not the only method that gets mirrored between static sounds and streamed sounds. Also, nothing prevents to introduce _new_ traits, which is basically what any kira consumer has to do to handle both kind of sounds in an agnostic fashion.
